### DISCUSSION TITLE
Where are grid shift files saved?

### DISCUSSION THREAD

**Discussion by TJPovey** [MaxRev-Dev/gdal.netcore#90] [Q&A] [answered]
Hi, I'm under the suspicion that a grid shift file is missing, more specifically uk_os_OSTN15_NTv2_OSGBtoETRS.tif.

This is due to coordinates being projected incorrectly from EPSG:27700 and to others (such as 4326 & 4979) using GDAL Net, but correct when using the gdal command line tool - gdaltransform along with other tools (epsg, fme, ordanance survey) which produce the correct result.

Is there a way to import the gridshift file into Gdal NET to see if this is causing the issue? Or can we specify Gdal to point to a directory to use these shift files by default? I can't seem to find where these files are saved, so I'm not sure where to place the tif file.

Thanks for your time,
Tom

## EDIT
Adding the below to the proj grid shift files did the trick. Do we know if these are meant to be included with the Gdal Net install?
```
Proj.Configure(@"C:\OSGeo4W\share\proj");
```

**Comment by MaxRev-Dev** [comment, marked as answer]
If you have a different directory for the proj data (grid shifts), you can include it when configuring PROJ search paths.
Of course, you can specify your proj data with `Proj.Configure(@"C:\OSGeo4W\share\proj");`.
By default, I do package only a small [`proj.db`](https://proj.org/resource_files.html#proj-db) file from proj build folder.
To keep packages neater and more concise other projections or newer proj-datumgrid are omitted. 

You can ensure that proj CRS is set by checking the output of 

   ``` csharp
var targetCrs = new SpatialReference(string.Empty);
var isSet = targetCrs.SetWellKnownGeogCS("EPSG:27700");
if (isSet == 0) {
// your projection
} else { 
// this CRS is not defined
}
```

But you can import a CRS from WKT ([EPSG source](https://epsg.io/27700.esriwkt)).
https://github.com/MaxRev-Dev/gdal.netcore/blob/d877176511ccb816acfa638a09cea81ac1aadabf/tests/MaxRev.Gdal.Core.Tests.XUnit/ProjTests.cs#L22-L28

**Reply by TJPovey** [reply]
That's great thanks for the info @MaxRev-Dev

As a note, the ``Proj.Configure`` method states that it takes an array of *additional* search paths, but when I log the search paths using:
``OSGeo.OSR.Osr.GetPROJSearchPaths()``
The only search path is the one set using ``Proj.Configure``. Without setting additional search paths, the ``OSGeo.OSR.Osr.GetPROJSearchPaths()`` method returns the path to the directory containing the proj.db packaged by Gdal Net. Not much of an issue as this probably removes any conflicts between the two, but it does state additional search paths and not that it replaces others.

I've also noticed you can access the grids via a CDN which I'll look into.

Thanks for your time

**Reply by MaxRev-Dev** [reply]
The main purpose of Proj.Configure is to find a proj.db (that one within the package). I did not use any additional file grids to configure proj as there was no need. But that's a good point to provide all the search paths to the proj. I'll make a patch for this.

**Reply by TJPovey** [reply]
Understood, thanks

**Reply by MaxRev-Dev** [reply]
You can use the official ESPG to find and export a specific CRS for your needs.
For example, https://epsg.io/32731. There are a lot of valuable formats like WKT, JSON, PostGIS, etc.
